### PR TITLE
Don't skip suppressors with /skipAnalyzers

### PR DESCRIPTION
**Pull request by mavasani**
Skipping suppressors can actually be a breaking change as an unsuppressed warning that could have been suppressed by a suppressor can be escalated to an error with /warnaserror. `skipAnalyzers` flag was only designed to skip diagnostic analyzers, so this PR ensures the same by never skipping suppressors.

**NOTE:** First 2 commits in the PR are just cherry-picks from https://github.com/dotnet/roslyn/pull/54915, which fix and unskip existing DiagnosticSuppressor unit tests.

### CODE REVIEW

File: src/Compilers/Core/Portable/CommandLine/CommandLineArguments.cs [hunk 1]
```diff
@@ -524,12 +524,14 @@ public IEnumerable<AnalyzerReference> ResolveAnalyzerReferences(IAnalyzerAssembl
                 }
             }
 
-            // All analyzer references are registered now, we can start loading them:
+            // If we are skipping analyzers, ensure that we only add suppressors.
+            bool shouldIncludeAnalyzer(DiagnosticAnalyzer analyzer) => !skipAnalyzers || analyzer is DiagnosticSuppressor;
```
Comment (333fred) [inline src/Compilers/Core/Portable/CommandLine/CommandLineArguments.cs:528]: Nit: Consider putting this at the end of the function, or just inlining as a lambda.

**Review by 333fred — APPROVED** [review]
LGTM (commit 4), with one minor formatting feedback.

File: src/Compilers/Core/Portable/CommandLine/CommandLineArguments.cs (hunk 1 above)
Comment (jcouv) [reply src/Compilers/Core/Portable/CommandLine/CommandLineArguments.cs:528]: Yes, please move local function toward the end of the method.

**Review by jcouv — COMMENTED** [review]
Done with review pass (iteration 4)

File: src/Compilers/Core/Portable/CommandLine/CommandLineArguments.cs (hunk 1 above)
Comment (mavasani) [reply src/Compilers/Core/Portable/CommandLine/CommandLineArguments.cs]: Done - sorry for the delay. @jcouv can you please take another look?

**Comment by mavasani** [comment]
@jcouv Can you take another look?

**Pull request merged**